test/new-e2e/utils/e2e/client/agent.go

```diff
@@ -0,0 +1,30 @@
+package client
+
+import (
+	"github.com/DataDog/test-infra-definitions/datadog/agent"
+)
+
+var _ stackInitializer = (*Agent)(nil)
+
+// A client Agent that is connected to an agent.Installer defined in test-infra-definition.
+type Agent struct {
+	*UpResultDeserializer[agent.InstallerData]
+	*sshClient
+}
+
+// Create a new instance of Agent
+func NewAgent(installer *agent.Installer) *Agent {
+	agent := &Agent{}
+	agent.UpResultDeserializer = NewUpResultDeserializer(installer.GetClientDataDeserializer(), agent.init)
+	return agent
+}
+
+func (agent *Agent) init(auth *Authentification, data *agent.InstallerData) error {
+	var err error
+	agent.sshClient, err = newSSHClient(auth, &data.Connection)
+	return err
+}
+
+func (agent *Agent) Status() (string, error) {
+	return agent.sshClient.Execute("sudo datadog-agent status")
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
🎉